mycore.org/content/de/documentation/basics/basics_mcr_ocfl.md

```diff
@@ -0,0 +1,312 @@
+
+Link zur Spezifikation: [0003-hash-and-id-n-tuple-storage-layout.md <sup><i class="fas fa-external-link-alt"></i></sup>](https://github.com/OCFL/extensions/blob/main/docs/0003-hash-and-id-n-tuple-storage-layout.md)
+
+#### Hash Generation
+
+Die Hashes im Implementierten *0003-hash-and-id-n-tuple-storage-layout* werden mit Sha-256 Generiert, 3 Tuple aus jeweils 3 Stellen generieren dann den Pfad zum Objekt.\
+Die Bildung des Hashes wird mit dem Ursprünglichen Dateinamen gemacht, welcher dann vor dem Speichern sicher für die Ordnerstruktur dank Escape-Sequenzen gemacht wird.
+
+{{<mcr-table id="repository-provider-list" class="table" style="" col-styles="">}}
+| Dateiname                          | Tuple 1 | Tuple 2 | Tuple 3 | Rest des Hashes                                         |
+| :--------------------------------- | :-----: | :-----: | :-----: | :------------------------------------------------------ |
+| derivate:Project_derivate_00000101 | a32     | 302     | e6c     | f89914a40e3656dac10c66586f0a8db3da22220030b7ba73da47350 | 
+| derivate:Project_derivate_00000109 | 484     | 67d     | d9f     | e493cdd42659c08844a487f1b9b21d8a229404b247e1cabba4f51c2 | 
+| derivate:Project_derivate_00000110 | f10     | 8d6     | 503     | 58f00e0affef633b0f618328a214b96fbf29f87f29ff7387e23a247 | 
+| derivate:Project_derivate_12345678 | 71d     | 94b     | ed9     | 219276d96aa6f0a6af1c31dfc203b3a8100f57b6a677feb222edc4b | 
+| doctype:Project_doctype_00000101   | 700     | 702     | 393     | 015ec5378a4008acbfa4239b89d3279f75cd7bf0d65227cedf4fdaa | 
+| doctype:Project_doctype_00000109   | 281     | dfd     | ee7     | 83f19825711d3a2f1544023f445539b6d666eccc1ac4271e657de73 | 
+| doctype:Project_doctype_00000110   | cf3     | 8d8     | 0a3     | c278bc2a78d094dc592052af2333d97f949872f3ff21ad51de7761a | 
+| doctype:Project_doctype_12345678   | 9d4     | 759     | 752     | 65d474fe6bcd7cfa9756cd06c690ae102f1ba5bfaf51294d96f418c | 
+| mcrclass:rfc5646                   | d32     | 4be     | d1c     | a6a89d0ed26bddcaaca66e50f57dd77b908c90ada7c21ba4489a26d | 
+| mcruser:editor1A@local             | 1a5     | ec9     | a72     | 9b5220c75033f21e46b594e22ec52d1f89b238072bf7488b6f32a07 | 
+| mcracl:rules                       | e64     | 6f0     | 669     | a8ae91be6c36beb30ba8d596f52682ecc2bc5a124b3d21d54967077 | 
+| mcrweb:pages                       | 5cd     | 8a6     | 495     | 343a81793d771c3e083674872d2763d7a3a112c251781e1052d5bba | 
+{{</mcr-table>}}
+
+#### Ordnerstruktur
+
+```text {linenos=table}
+[storage_root]
+├── 0=ocfl_1.0
+├── ocfl_layout.json
+├── extensions
+│   └── 0003-hash-and-id-n-tuple-storage-layout
+│       └── config.json
+├── a32
+│   └── 302
+│       └── e6c
+│           └── derivate%3aProject_derivate_00000101
```

Comment:
Konsistent wäre `mcrderivate%3a:Project_derivate_00000101`
Das hätte mir aber auch schon mal eher auffallen können

Reply:
Dies ist schon so korrekt, da `%3a` ja schon `:` ist, einfach nur die maskierte (bzw. escaped) version.
Der Grund warum das MyCoRe Storage Layout dies nicht besitzt ist, dass die IDs schon "rein" sind, da man diese nicht selber angeben kann sondern zugewiesen werden.

Reply:
Bei Windows als auch auf Linux macht "%" keine Probleme, Mac kann ich nicht testen, aber in Recherchen sieht es nicht so aus als gibt es damit Probleme.

Reply:
Na gut, dann hoffen wir mal das das so bleibt - Schließlich soll man das OCFL-Repository ja auch in 1000 Jahren noch lesen können. ;-) Ich bin ja eh Verfechter des "MyCoRe-Storage-Layouts" - da sind wir auf der "sicheren Seite".